modules/host/negotiate.go

```diff
@@ -165,16 +165,13 @@ func (h *Host) considerRevision(txn types.Transaction, obligation contractObliga
 	return nil
 }
 
-// rpcUpload is an RPC that negotiates a file contract. Under the new scheme,
-// file contracts should not initially hold any data.
-func (h *Host) rpcUpload(conn net.Conn) error {
-	// Check that the host has grabbed an address from the wallet.
-	if h.UnlockHash == (types.UnlockHash{}) {
-		return errors.New("host needs an address; have you properly announced?")
-	}
-
-	// allow 1 minute for contract negotiation
-	conn.SetDeadline(time.Now().Add(1 * time.Minute))
+// negotiateContract negotiates an initial file contract with a renter, and
+// adds the metadata to the host's obligation set. The merkleRoot and filename
+// arguments are provided to make negotiateContract usable with both rpcUpload
+// and rpcRenew.
+func (h *Host) negotiateContract(conn net.Conn, merkleRoot crypto.Hash, filename string) error {
+	// allow 5 minutes for contract negotiation
+	conn.SetDeadline(time.Now().Add(5 * time.Minute))
```

Comment:
The hostdb actually only allows 30 seconds for contract negotiation, because more than that usually means the connection will time out. But I agree, it would be useful to know the average time required for each RPC.